In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from datetime import datetime

# REPORT DATE
report_date = datetime.today().strftime('%B %Y')
print(f'Monthly Report: {report_date}')

Monthly Report: July 2026


In [15]:
# LOAD DATA
df = pd.read_csv('commodity_data.csv', index_col='Date', parse_dates=True)
spread = df['Brent'] - df['WTI']

# KEY METRICS
current_spread = spread.iloc[-1]
spread_mean_30 = spread.rolling(30).mean().iloc[-1]
spread_std_30 = spread.rolling(30).std().iloc[-1]
zscore = (current_spread - spread_mean_30) / spread_std_30
# DRAWDOWN FROM 2024
spread_2024 = spread['2024-01-01':]
rolling_max_2024 = spread_2024.cummax()
current_drawdown_2024 = (spread_2024 - rolling_max_2024).iloc[-1]
# VOLATILITY
vol_30 = spread.rolling(30).std().iloc[-1]
vol_90 = spread.rolling(90).std().iloc[-1]

# DRAWDOWN
rolling_max = spread.cummax()
current_drawdown = (spread - rolling_max).iloc[-1]

# DAILY CHANGES
daily_change = df.iloc[-1] - df.iloc[-2]
daily_change_pct = ((df.iloc[-1] - df.iloc[-2]) / df.iloc[-2] * 100).round(2)

print('Data loaded successfully')

Data loaded successfully


In [16]:
# PRINT MONTHLY REPORT
print('=' * 60)
print(f'COMMODITY MARKETS MONTHLY REPORT — {report_date}')
print('=' * 60)

print('\n📊 BRENT-WTI SPREAD')
print(f'  Current Spread:        ${current_spread:.2f}')
print(f'  30 Day Average:        ${spread_mean_30:.2f}')
print(f'  Current Drawdown:      ${current_drawdown_2024:.2f}')
print(f'  Z Score:               {zscore:.2f}')
print(f'  Signal:                {"Buy" if zscore < -2 else "Sell" if zscore > 2 else "Neutral"}')

print('\n📈 VOLATILITY')
print(f'  30 Day Volatility:     ${vol_30:.2f}')
print(f'  90 Day Volatility:     ${vol_90:.2f}')
print(f'  Regime:                {"High" if vol_30 > vol_90 * 1.2 else "Low" if vol_30 < vol_90 * 0.8 else "Normal"}')

print('\n💹 DAILY PERFORMANCE')
for commodity in df.columns:
    change = daily_change_pct[commodity]
    arrow = '▲' if change > 0 else '▼'
    print(f'  {commodity:<12} {arrow} {change:>6.2f}%')

notes = f"""
{'=' * 60}
ANALYST NOTES — {report_date}
{'=' * 60}

MARKET OVERVIEW:
  - Major macro events this month?
  - Energy market response?
  - Geopolitical developments?

SPREAD ANALYSIS:
  - Spread vs 30 day average: ${current_spread:.2f} vs ${spread_mean_30:.2f}
  - Z Score signal: {"Buy" if zscore < -2 else "Sell" if zscore > 2 else "Neutral"}
  - What is driving the spread?

VOLATILITY:
  - Current regime: {"High" if vol_30 > vol_90 * 1.2 else "Low" if vol_30 < vol_90 * 0.8 else "Normal"}
  - Any spikes this month?

FORWARD CURVE:
  - Contango or backwardation?
  - Curve shape vs last month?

KEY RISKS:
  - What could move the spread significantly?
  - Scheduled events to watch?

OUTLOOK:
  - Base case next 30 days?
  - Key levels to watch?
{'=' * 60}
"""

print(notes)

COMMODITY MARKETS MONTHLY REPORT — July 2026

📊 BRENT-WTI SPREAD
  Current Spread:        $3.15
  30 Day Average:        $3.56
  Current Drawdown:      $-13.82
  Z Score:               -0.29
  Signal:                Neutral

📈 VOLATILITY
  30 Day Volatility:     $1.40
  90 Day Volatility:     $3.88
  Regime:                Low

💹 DAILY PERFORMANCE
  Brent        ▼  -2.55%
  WTI          ▼  -2.29%
  NatGas       ▲   7.98%
  Gold         ▲   0.21%
  Silver       ▲   0.80%
  Copper       ▼  -0.65%
  Platinum     ▲   1.09%
  Sugar        ▼  -2.26%
  Coffee       ▲   2.48%
  Cotton       ▲   4.32%
  Wheat        ▲   0.55%
  Corn         ▲   7.03%
  Soybeans     ▲   2.91%
  DXY          ▲   2.93%
  SP500        ▼  -0.22%

ANALYST NOTES — July 2026

MARKET OVERVIEW:
  - Major macro events this month?
  - Energy market response?
  - Geopolitical developments?

SPREAD ANALYSIS:
  - Spread vs 30 day average: $3.15 vs $3.56
  - Z Score signal: Neutral
  - What is driving the spread?

VOLATILITY:


In [17]:
with open('monthly_notes_june_2026.txt', 'w') as f:
    f.write(f'{"=" * 60}\n')
    f.write(f'COMMODITY MARKETS MONTHLY REPORT — {report_date}\n')
    f.write(f'{"=" * 60}\n\n')
    f.write(f'BRENT-WTI SPREAD\n')
    f.write(f'  Current Spread:     ${current_spread:.2f}\n')
    f.write(f'  30 Day Average:     ${spread_mean_30:.2f}\n')
    f.write(f'  Current Drawdown:   ${current_drawdown_2024:.2f}\n')
    f.write(f'  Z Score:            {zscore:.2f}\n')
    f.write(f'  Signal:             {"Buy" if zscore < -2 else "Sell" if zscore > 2 else "Neutral"}\n\n')
    f.write(f'VOLATILITY\n')
    f.write(f'  30 Day:             ${vol_30:.2f}\n')
    f.write(f'  90 Day:             ${vol_90:.2f}\n')
    f.write(f'  Regime:             {"High" if vol_30 > vol_90 * 1.2 else "Low" if vol_30 < vol_90 * 0.8 else "Normal"}\n\n')
    f.write('DAILY PERFORMANCE\n')
    for commodity in df.columns:
        change = daily_change_pct[commodity]
        arrow = '▲' if change > 0 else '▼'
        f.write(f'  {commodity:<12} {arrow} {change:>6.2f}%\n')
    f.write('\n')
    f.write('ANALYST NOTES\n')
    f.write('  Market Overview:    [ADD NOTES]\n')
    f.write('  Key Risks:          [ADD NOTES]\n')
    f.write('  Outlook:            [ADD NOTES]\n')

print('Monthly notes saved to monthly_notes_july_2026.txt')

Monthly notes saved to monthly_notes_july_2026.txt


## June 2026 — Analyst Commentary

### Market Overview
- Broad energy selloff today, Brent -2.55% WTI -2.29%
- NatGas +7.98% significant single day move
- Dollar strengthened significantly, DXY +2.93%
- Grains rallied strongly — Corn +7.03%, Cotton +4.32%

### Spread Analysis
- Spread at $3.15, slightly below 30 day average of $3.56
- Z score -0.29 — neutral, no actionable signal
- Ceasefire announced June 17, risk premium has largely deflated

### Volatility
- Currently in low regime post ceasefire
- 30 day vol $1.40 well below 90 day $3.88
- Market calming but NatGas move worth monitoring

### Key Risks
- Ceasefire breakdown — re-escalation would spike Brent immediately
- NatGas move needs monitoring for follow through
- DXY strength could continue weighing on commodities

### Outlook
- Base case spread stays around $3.00-$3.50 near term
- Mean reversion model suggests drift back toward $3.56 average
- Watch NatGas and grain moves for continuation